doc/tutorials/contributing_code.rst

```diff
@@ -0,0 +1,268 @@
+
+
+Create a new branch
+===================
+Create a new branch ``new-uid`` for your changes (you can choose any name
+that you want instead). Any changes made in this branch will be specific to
+it and won't affect the main copy (the ``master`` branch) of the code::
+
+  $ git checkout -b new-uid
+
+
+Write tests for your changes
+============================
+If a change is to be accepted into *pydicom* it usually has to include tests.
+For bug fixes you should write a regression test that reproduces the bug.
+For new features you'll need to include tests that ensure the features
+work as intended.
+
+Let's say we wanted to add a new `pre-defined UID
+<https://pydicom.github.io/pydicom/dev/reference/uid.html#predefined-uids>`_
+to *pydicom* with a value of ``1.2.3.4.500``. We'd first add a new test at the
+bottom of `test_uid.py
+<https://github.com/pydicom/pydicom/blob/master/pydicom/tests/test_uid.py>`_::
+
+  def test_new_uid():
+      """Test uid.NewDefinedUID."""
+      from pydicom.uid import NewDefinedUID
+      assert '1.2.3.4.500' == NewDefinedUID
+
+.. note::
+
+   If you've never had to write tests before they can seem pretty daunting,
+   especially if you're also learning how to use pytest from scratch. You may
+   find the following resources useful:
+
+   * Take a look at the
+     `existing test suite <https://github.com/pydicom/pydicom/tree/master/pydicom/tests>`_
+     and see how the tests are written. There are examples for writing
+     `a single test <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_rle_pixel_data.py#L137>`_,
+     `a group of related tests <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_dataelem.py#L27>`_,
+     `testing for exceptions <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_handler_util.py#L834>`_,
+     `capturing log output <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_config.py#L28>`_,
+     `testing for warnings <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_pillow_pixel_data.py#L452>`_,
+     and running
+     `parametrized tests <https://github.com/pydicom/pydicom/blob/73cffe3151915b53a18b521656680d819e7e1a18/pydicom/tests/test_rle_pixel_data.py#L215>`_.
+   * Dive Into Python has a very nice `section on unit testing
+     <https://diveinto.org/python3/unit-testing.html>`_ (however it uses
+     ``unittest`` instead of pytest).
+   * The `pytest documentation <https://docs.pytest.org/en/latest/example/index.html>`_
+     may also be helpful
+
+   In the worst-case scenario, if you can't figure out how to write a test
+   for something, then once you've created a pull-request (to be discussed
+   a bit later) add a comment asking for help.
+
+Since we haven't made any modification to the actual source code, when we
```

Comment:
The note above disrupts reading a bit - I would place it below the paragraph so that the actual text can be seen as one block.

Reply:
I'll move the note up above "Let's say..."